frontend/src/pages/Index.js

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from "react";
+import { batch, useDispatch, useSelector } from "react-redux";
+import { useHistory } from "react-router-dom";
+
+import thoughts from "../reducers/thoughts";
+import user from "../reducers/user";
+
+import "./IndexStyle.css";
+
+const Index = () => {
+  const [message, setMessage] = useState("");
+  const accessToken = useSelector((store) => store.user.accessToken);
+  const thoughtsItems = useSelector((store) => store.thoughts.items);
+
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const API_THOUGHTS = "https://week20-project-auth.herokuapp.com/thoughts";
+
+  // This useEffect will check is the accessToken in the redux state is null. If it is null then the user will be sent to /login page.
+  useEffect(() => {
+    if (!accessToken) {
+      history.push("/login");
+    }
+  });
+
+  // This useEffect will send a GET request to to the backend and the backend will send a response back with all of the thoughts that are in the database
+  useEffect(() => {
+    const API_THOUGHTS = "https://week20-project-auth.herokuapp.com/thoughts";
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: accessToken,
+      },
+    };
+
+    fetch(API_THOUGHTS, options)
+      .then((res) => res.json())
+      .then((data) => {
+        if (data) {
+          batch(() => {
+            dispatch(thoughts.actions.setThoughts(data));
+            dispatch(thoughts.actions.setErrors(null));
+          });
+        } else {
+          dispatch(thoughts.actions.setErrors(data));
+        }
+      });
+  }, [message, accessToken, dispatch]);
+
+  const onLogout = (e) => {
+    e.preventDefault();
+    dispatch(user.actions.setAccessToken(""));
+  };
+
+  const onButtonClick = (e) => {
```

Comment:
This post function doesn´t seem to work properly, new messages do not appear in the list immediately but only after posting another message or reloading the page.